packages/analytics-browser/src/browser-client.ts

```diff
@@ -174,226 +175,29 @@ export class AmplitudeBrowser extends AmplitudeCore<BrowserConfig> {
- * revenue(rev);
- *
- * // alternatively, this tracking method is awaitable
- * const result = await revenue(rev).promise;
- * console.log(result.event); // {...}
- * console.log(result.code); // 200
- * console.log(result.message); // "Event tracked successfully"
- * ```
- */
-export const revenue = returnWrapper(client.revenue.bind(client));
-
-/**
- * Returns current user ID.
- *
- * ```typescript
- * const userId = getUserId();
- * ```
- */
-export const getUserId = client.getUserId.bind(client);
-
-/**
- * Sets a new user ID.
- *
- * ```typescript
- * setUserId('userId');
- * ```
- */
-export const setUserId = client.setUserId.bind(client);
-
-/**
- * Returns current device ID.
- *
- * ```typescript
- * const deviceId = getDeviceId();
- * ```
- */
-export const getDeviceId = client.getDeviceId.bind(client);
-
-/**
- * Sets a new device ID.
- * When setting a custom device ID, make sure the value is sufficiently unique.
- * A uuid is recommended.
- *
- * ```typescript
- * setDeviceId('deviceId');
- * ```
- */
-export const setDeviceId = client.setDeviceId.bind(client);
-
-/**
- * reset is a shortcut to anonymize users after they log out, by:
- *   - setting userId to `undefined`
- *   - regenerating a new random deviceId
- *
- * With an `undefined` userId and a completely new deviceId, the current user would appear as a brand new user in dashboard.
- *
- * ```typescript
- * reset();
- * ```
- */
-export const reset = client.reset.bind(client);
-
-/**
- * Returns current session ID.
- *
- * ```typescript
- * const sessionId = getSessionId();
- * ```
- */
-export const getSessionId = client.getSessionId.bind(client);
-
-/**
- * Sets a new session ID.
- * When settign a custom session ID, make sure the value is in milliseconds since epoch (Unix Timestamp).
- *
- * ```typescript
- * setSessionId(Date.now());
- * ```
- */
-export const setSessionId = client.setSessionId.bind(client);
-
-/**
- * Sets a new optOut config value. This toggles event tracking on/off.
- *
- *```typescript
- * // Stops tracking
- * setOptOut(true);
- *
- * // Starts/resumes tracking
- * setOptOut(false);
- * ```
- */
-export const setOptOut = client.setOptOut.bind(client);
-
-/**
- *  Sets the network transport type for events.
- *
- * ```typescript
- * // Use Fetch API
- * setTransport('fetch');
- *
- * // Use XMLHttpRequest API
- * setTransport('xhr');
- *
- * // Use navigator.sendBeacon API
- * setTransport('beacon');
- * ```
- */
-export const setTransport = client.setTransport.bind(client);
-
-/**
- * Flush and send all the events which haven't been sent.
- *
- *```typescript
- * // Send all the unsent events
- * flush();
- *
- * // alternatively, this tracking method is awaitable
- * await flush().promise;
- * ```
- */
-export const flush = returnWrapper(client.flush.bind(client));
+export const createInstance: CreateBrowserInstance = () => {
+  const client = new AmplitudeBrowser();
+  return {
+    init: amplitudePromise(client.init.bind(client)),
```

Comment:
Will exporting the methods grouped like this affect tree-shaking?

Reply:
Yes! Although, importing `init` and `track` imports the ~90% SDK that's why I don't feel as bad now.